packages/build-tools/src/utils/project.ts

```diff
@@ -21,6 +22,13 @@ export async function setup<TJob extends Job>(ctx: BuildContext<TJob>): Promise<
     if (ctx.env.NPM_TOKEN) {
       await createNpmrcIfNotExistsAsync(ctx);
     }
+    if (ctx.job.platform === Platform.IOS && ctx.env.EAS_BUILD_ENVIRONMENT === 'eas-build') {
+      // .xcode.env.local
+      const deleted = await deleteXcodeEnvLocalIfExistsAsync(ctx as BuildContext<Ios.Job>);
+      if (deleted) {
+        ctx.markBuildPhaseHasWarnings();
```

Comment:
I usually try to avoid functions that both do sth and return the status of that thing. Consider moving markBuildPhaseHasWarnings inside deleteXcodeEnvLocalIfExistsAsync.